src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/service/impl/MessageRejectionServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.service.impl;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.support.serializer.DeserializationException;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.service.MessageRejectionService;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.slack.SlackMessagingService;
+
+@Service
+public class MessageRejectionServiceImpl implements MessageRejectionService {
+
+    @Autowired
+    private ApplicationLogger logger;
+
+    @Autowired
+    private SlackMessagingService slackMessagingService;
+
+
+    @Override
+    public void handleRejectedMessage(Exception exception, ConsumerRecord<?, ?> datum) {
+        String errorMessageDetails = buildMessage(datum.topic(), datum.partition(), datum.offset());
+
+        if (exception.getCause() instanceof DeserializationException) {
+            sendToSlack(errorMessageDetails);
+        } else {
+            logger.error(String.format("Kafka message exception - %s - %s", exception.getMessage(), errorMessageDetails));
```

Comment:
if there is a non deserialize exception, it won't message slack. I'm thinking any errors that cause this method to be run should notify the support team?  one for @ICoxCH as well.

Reply:
agreed it should message slack, but i'd argue seperate story/pr?

Maybe a more generic message to be sent in that case (I can't think of any scenarios that would happen but wouldnt hurt to code for the worst)

Reply:
ok